flang/lib/Optimizer/Builder/HLFIRTools.cpp

```diff
@@ -259,8 +260,11 @@ hlfir::AssociateOp hlfir::genAssociateExpr(mlir::Location loc,
   }
   llvm::SmallVector<mlir::Value> lenParams;
   genLengthParameters(loc, builder, value, lenParams);
-  return builder.create<hlfir::AssociateOp>(loc, source, name, shape, lenParams,
-                                            fir::FortranVariableFlagsAttr{});
+  auto res = builder.create<hlfir::AssociateOp>(
+      loc, source, name, shape, lenParams, fir::FortranVariableFlagsAttr{});
+  if (attr)
+    res->setAttrs({*attr});
```

Comment:
I think it would be cleaner to add an op builder for AssociateOp which takes a NamedAttribute parameter (in the same way as this is done for fir::AllocaOp).

Reply:
I have broken something trying to fix this. Will work on it tomorrow.

Reply:
Trivial error once I looked at it in fresh light. :) [I had an assert the wrong way around - which caused some 50-something tests to fail].